WalletWasabi.Fluent/Models/UiTriggers.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Reactive;
+using System.Reactive.Linq;
+using ReactiveUI;
+using WalletWasabi.Blockchain.TransactionProcessing;
+using WalletWasabi.Fluent.Extensions;
+using WalletWasabi.Fluent.ViewModels.Wallets;
+using WalletWasabi.Wallets;
+
+namespace WalletWasabi.Fluent.Models;
+
+public class UiTriggers
+{
+	private readonly WalletViewModel _walletViewModel;
+	private readonly Wallet _wallet;
+
+	public UiTriggers(WalletViewModel walletViewModel)
+	{
+		_walletViewModel = walletViewModel;
+		_wallet = _walletViewModel.Wallet;
+	}
+
+	/// <summary>
+	/// Triggers on subscription and when a transaction to the wallet is processed.
+	/// </summary>
+	public IObservable<Unit> WalletRelevantTransactionProcessed =>
+		Observable.FromEventPattern(_wallet.TransactionProcessor, nameof(TransactionProcessor.WalletRelevantTransactionProcessed)).ToSignal().StartWith(Unit.Default);
+
+	/// <summary>
+	/// Triggers on subscription and when the USD exchange rate changed.
+	/// </summary>
+	public IObservable<Unit> UsdExchangeRateChanged => _wallet.Synchronizer.WhenAnyValue(x => x.UsdExchangeRate).ToSignal();
+
+	/// <summary>
+	/// Triggers on subscription and when the anon score target changed.
+	/// </summary>
+	public IObservable<Unit> AnonScoreTargetChanged => _walletViewModel.CoinJoinSettings.WhenAnyValue(x => x.AnonScoreTarget).ToSignal().Throttle(TimeSpan.FromMilliseconds(3100));
+
+	/// <summary>
+	/// Triggers on subscription and when a transaction is processed to the wallet or the USD exchange rate changed.
+	/// </summary>
+	public IObservable<Unit> BalanceUpdateTrigger => WalletRelevantTransactionProcessed.Merge(UsdExchangeRateChanged);
```

Comment:
@SuperJMN because of the merge it will trigger twice... Is there a better approach instead of `throttle` to make it only 1 trigger?

Reply:
What do you mean by "it will trigger twice"?

Reply:
For example the Balance tile is using this to get the sign when to update. When its subscribes it will get two sign instantly, one from `WalletRelevantTransactionProcessed` and one from `UsdExchangeRateChanged`. So it will update twice which is unnecessary.

I would use Throttle to have only one sign, but I know you know RX better so this is why I am asking if there is a better approach.